### DISCUSSION TITLE
Passing build settings to scikit-build-core when using the hatchling plugin

### DISCUSSION THREAD

**Discussion by jcfr** [scikit-build/scikit-build-core#1467] [General]
Working on using `scikit-build-core` in a project already using hatchling[^1], I realized that passing CMake option using command-line argument does not seem to be supported.

For example, trying to pass option to `build` like using argument like `-Ccmake.define.VARNAME=VALUE` is not working as expected:

```
python -m build -wn -v \
  -Cbuild-dir="build/{wheel_tag}" \
  -Ccmake.define.CMAKE_PREFIX_PATH=/tmp/foo-install/ .
```

This is potentially related to the following issue:
* https://github.com/pypa/hatch/issues/1072

Next:
* Create a minimal reproducer example

[^1]: https://scikit-build-core.readthedocs.io/en/latest/plugins/hatchling.html

**Comment by henryiii** [comment]
I don't think there's a way for hatchling to provide the config-settings to us (or setuptools, for that matter).
